src/ngResource/resource.js

```diff
@@ -588,14 +657,23 @@ angular.module('ngResource', ['ng']).
               }
             });
 
+            if (!isInstanceCall && action.cancellable) {
+              var deferred = $q.defer();
+              httpConfig.timeout = deferred.promise;
+              value.$cancelRequest = deferred.resolve;
```

Comment:
Do we need to worry about memory leaks if this promise gets attached to the http request?